modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/ClickGoalPicker/ClickGoalPicker.es.js

```diff
@@ -370,25 +370,20 @@ function OverlayContainer({allowEdit, root}) {
 
 	useEventListener('keydown', handleKeydown, true, document);
 
-	const handleMouseDown = useCallback(() => {
-		mousedownRef.current = true;
-	}, []);
-
-	const handleMouseUp = useCallback(
+	const handleClick = useCallback(
 		(event) => {
-			if (mousedownRef.current === true) {
+			if (clickRef.current === true) {
 				event.preventDefault();
 				stopImmediatePropagation(event);
 				dispatch({type: 'deactivate'});
 			}
 
-			mousedownRef.current = false;
+			clickRef.current = !clickRef.current;
 		},
 		[dispatch]
 	);
 
-	useEventListener('mousedown', handleMouseDown, false, document);
-	useEventListener('mouseup', handleMouseUp, false, document);
+	useEventListener('click', handleClick, false, document);
```

Comment:
So from what I remember, I think changing this to a `click` event will cause a blur event and will cause the input to lose focus and then the element is selected but not highlighted.

Reply:
To test this bug, remove the `if ()` wrapper in the event handler and see how clicking outside of the input and hitting enter in the input have two different results.

Reply:
> So from what I remember, I think changing this to a click event will cause a blur event and will cause the input to lose focus and then the element is selected but not highlighted.

Yes, that happened with click event and no ref. In this case I used click event and ref. I mixed both concepts. I don't know if that is correct or not. With this change, if the user enters an ID in the input and hits enter and then wants to change it using the highlighted ones, the first click would lose the focus in the input and the second will be 1) cancel edition if the user clicks in a non-clickable element; 2) open the clickable element to select as target (popover + topper box view).